google-cloud-firestore/lib/google/cloud/firestore/watch/listener.rb

```diff
@@ -119,6 +120,8 @@ def stopped?
 
           def send_callback query_snp
             @callback.call query_snp
+          rescue StandardError => e
+            @parent.error! e
```

Comment:
@schmidt-sebastian The original implementation (added in #2202) did *not* rescue errors from user callbacks, but we decided to add it in this PR with the reasoning that the listener should be allowed to continue processing as long as the callback errors are reported to the user via `on_error` and `last_error`. Does this sound correct to you?

Reply:
Once the stream errors with a non-retryable error, the user will have to retry the operation. There will be no more updates from the stream. With that in mind, `on_error` should likely be treated as a final state.

Reply:
OK, I will remove this block, allowing the error to be handled in lines 297-8 below, which will raise the error, ending the listener thread.

Reply:
@schmidt-sebastian Maybe there was a misunderstanding. In this case, I was asking about an error originating in the user-provided callback. The `rescue` above currently catches an error originating in the user-provided callback, reports it, and continues the listening loop. Is that OK with you?

Reply:
That's fine.